### PR TITLE
3668 fhir patch operation returning 500 when patching in a transaction with a resource query in property request.url

### PR DESCRIPTION
**Pull request by epeartree**
**What Was Done**
1. A test called "testFhirPatch_TransactionWithSearchParameter" was created to address the issue
2. Adding a trivial fix to the issue
3. A change log was created regarding the modification

### CODE REVIEW

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3668-fhir-patch-operation-returning-500.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+type: fix
+issue: 3668
+title: "Fhir patch operation returning 500 when patching in a transaction with a resource query in property request.ur.l"
```
Comment (tadgh) [inline hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3668-fhir-patch-operation-returning-500.yaml:4]:
```suggestion
title: "Fhir patch operation returning 500 when patching in a transaction with a resource query in property request.url."
```

File: hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/r4/PatchProviderR4Test.java [hunk 2]
```diff
@@ -110,6 +110,47 @@ public void testFhirPatch_Transaction() throws Exception {
...
+		HttpPost post = new HttpPost(ourServerBase);
+		String encodedRequest = myFhirContext.newJsonParser().setPrettyPrint(true).encodeResourceToString(input);
+		ourLog.info("Request:\n{}", encodedRequest);
+
+		post.setEntity(new StringEntity(encodedRequest, ContentType.parse(Constants.CT_FHIR_JSON_NEW+ Constants.CHARSET_UTF8_CTSUFFIX)));
+		try (CloseableHttpResponse response = ourHttpClient.execute(post)) {
+			assertEquals(200, response.getStatusLine().getStatusCode());
+			String responseString = IOUtils.toString(response.getEntity().getContent(), StandardCharsets.UTF_8);
+			assertThat(responseString, containsString("\"resourceType\":\"Bundle\""));
+		}
```
Comment (tadgh) [inline hapi-fhir-jpaserver-test-utilities/src/test/java/ca/uhn/fhir/jpa/provider/r4/PatchProviderR4Test.java:148]: Suggestion: Rewrite this HTTP POST to just use the available IGenericClient

**Review by tadgh — APPROVED** [review]
Assuming this passes all test cases, I've got minor suggestions, and then happy to merge.

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3668-fhir-patch-operation-returning-500.yaml (hunk 1 above)
Comment (Qingyixia) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_1_0/3668-fhir-patch-operation-returning-500.yaml:4]: Sure, I will make the changes right now! Thank you.

**Pull request merged**